Yank/multistate/multistateanalyzer.py

```diff
@@ -1726,7 +1727,7 @@ def extract_decorrelated(cached_dict, dtype, unit):
                                                                 analysis_particles_only=True)
 
             for replica_idx, sampler_state in enumerate(sampler_states):
-                sliced_sampler_state = sampler_state[original_restrained_atom_indices]
+                sliced_sampler_state = sampler_state[reduced_restrained_atom_indices]
```

Comment:
I believe this won't work with the current implementation. Assume your solute-only sampler states covers particles [1000-2000], and in the simulation you're restraining [1500-1600] with [1900-2000]. I think this will re-compute the restraint distances and energies based on the positions of atoms [1000-1200].

It might be worth writing a regression test for this if you manage to encapsulate the interesting bit into a separate static function.

Reply:
I'll look into this when I get a chance - thanks!